modules/node-shared/src/main/scala/org/tessellation/node/shared/http/routes/SnapshotRoutes.scala

```diff
@@ -40,78 +43,102 @@ final case class SnapshotRoutes[F[_]: Async: Hasher: KryoSerializer, S <: Snapsh
   implicit def binaryAndJsonEncoders[A <: AnyRef: Encoder]: List[EntityEncoder[F, A]] =
     List(BinaryCodec.encoder[F, A], CirceEntityEncoder.circeEntityEncoder[F, A])
 
+  private def badRequestNodeNotReady(currentState: NodeState): F[Response[F]] =
+    BadRequest(("message" ->> s"Node should be in Ready state before fetching. Current state: $currentState") :: HNil)
+
   protected val httpRoutes: HttpRoutes[F] = HttpRoutes.of[F] {
     case GET -> Root / "latest" / "ordinal" =>
-      import org.http4s.circe.CirceEntityCodec.circeEntityEncoder
-
-      snapshotStorage.headSnapshot.map(_.map(_.ordinal)).flatMap {
-        case Some(ordinal) => Ok(("value" ->> ordinal.value.value) :: HNil)
-        case None          => NotFound()
+      nodeState.flatMap {
+        case NodeState.Ready =>
+          snapshotStorage.headSnapshot.map(_.map(_.ordinal)).flatMap {
+            case Some(ordinal) => Ok(("value" ->> ordinal.value.value) :: HNil)
+            case None          => NotFound()
+          }
+        case currentState => badRequestNodeNotReady(currentState)
```

Comment:
please don't use a check for a specific node state. use `is(ready)` function instead. for `Ready` state it's a simple check, but for node being in cluster, it's more complex as there are multiple states to check against. having it done via the fn, we keep it open for extending (which is one of the SOLID rules)